File: API/APIMeow/ViewModels/CreateMetasViewModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace APIMeow.ViewModels
{
    public class CreateMetasViewModel
    {
 public string Nome { get; set; }
    public decimal GastoLimite { get; set; }
    public int QtsMoedas { get; set; }
    public DateTime DataCriacao { get; set; }
    public DateTime DataTermino { get; set; }
    public char Feita { get; set; } = 'N';
    public int IdUsuario { get; set; }
   
    public int IdClassificacao { get; set; }

    }
}

File: front_meow/lib/model/CreateMetasViewModel.dart
class CreateMetasViewModel {
  String nome;
  double gastoLimite;
  int idClassificacao;
  DateTime dataCriacao;
  DateTime dataTermino;
  String feita;

  CreateMetasViewModel({
    required this.nome,
    required this.gastoLimite,
    required this.idClassificacao,
    required this.dataCriacao,
    required this.dataTermino,
    required this.feita,
  });

  Map<String, dynamic> toJson() => {
    'nome': nome,
    'gastoLimite': gastoLimite,
    'idClassificacao': idClassificacao,
    'dataCriacao': dataCriacao.toIso8601String(),
    'dataTermino': dataTermino.toIso8601String(),
    'feita': feita,
  };

  factory CreateMetasViewModel.fromJson(Map<String, dynamic> json) {
    return CreateMetasViewModel(
      nome: json['nome'],
      gastoLimite: (json['gastoLimite'] as num).toDouble(),
      idClassificacao: json['idClassificacao'],
      dataCriacao: DateTime.parse(json['dataCriacao']),
      dataTermino: DateTime.parse(json['dataTermino']),
      feita: json['feita'],
    );
  }
}
